### DISCUSSION TITLE
html inline CSS properties into the style attribute

### DISCUSSION THREAD

**Discussion by SilvaQ** [rehypejs/rehype#98] [Ideas] [answered]
Is it possible that we could combine the power of juice's library with our Rehype line and offer an ast extension?

Markdown to HTML special scenes need this ability, if we can customize it better!

Juice project address:
https://github.com/Automattic/juice
> Given HTML, juice will inline your CSS properties into the style attribute.

**Comment by ChristianMurphy** [comment, marked as answer]
Could `juice` be used with Rehype? Yes
It works on raw HTML, so Rehype would need to stringify then re-parse to be integrated in the pipeline, or `juice` could be applied after Rehype is done.

> offer an ast extension

It could be made a plugin with the stringify and re-parse approach noted above.
`juice` doesn't support ASTs and is unlikely to change anytime soon, given that it is not currently maintained https://github.com/Automattic/juice/issues/399

That said, if you wanted to create an AST based alternative, you could.
In theory https://www.npmjs.com/package/hast-util-select could be used as an alternative to jQuery/Cheerio selectors to re-implement the functionality.
